doc/debugging.md

```diff
@@ -0,0 +1,94 @@
+* The -i parameter specifies the private key that corresponds to the public key provided during the Discovery ISO generation.  
+* Replace <identity_file> with the actual path to your private key.
+
+**4. Unable to SSH into the Agent Machine through the network?**  
+If SSH fails, try accessing the machine directly using:
+* Physical access
+* Out-of-band management
+* A virtual machine console
+
+Once logged in:
+
+* The machine's IP address should be displayed in the CLI if a network interface is available.
+* Verify that the previously used IP address is correct and that Steps 1 and 2 were completed successfully.
+* Follow common troubleshooting steps such as:
+  1. Ping the agent from your machine.
+  2. Check firewall rules to ensure traffic to the agent is allowed.
+After resolving any connectivity issues, retry Step 3.
+
+## Now, depending on your case, proceed with the additional steps
+
+### Case: Unable to Access the Login Page
+
+**1. Verify That the Agent Components Are Running**  
+
+Run the following command to check running containers:
+
+`podman ps`  
+
+Expected output should include running containers similar to:
+
+```
+| CONTAINER ID | IMAGE                                               | COMMAND               | CREATED        | STATUS        | PORTS | NAMES          |
+| cc0a71a37c1b | quay.io/kubev2v/forklift-validation:release-v2.6.4  | run --server /usr...  | 2 minutes ago | Up 31 minutes |       | opa            |
+| 70ad0a7cbdc5 | quay.io/kubev2v/migration-planner-agent:latest      | -config /agent/co...  | 2 minutes ago | Up 31 minutes |       | planner-agent  |
+```
+
+If the agent components are not running, try restarting the agent VM.
+
+**2. Use the agent UI**  
+Navigate to: **`https://<machine-ip>:3333/login`**. You should see a login form asking for Vmware credentials. 
+If the site isn't reachable:  
+1. Ensure that traffic to port 3333 is allowed for the destination <machine-ip> in the firewall. 
+2. If the issue persists restart the agent VM and try again.
+
+### Case: Source Not Appearing in the Migration Assessment Service
+
+**1. Verify the Connection to the Service** 
+
+The agent exposes a `/status` endpoint, providing information to the agent UI and components regarding:
+
+* The status of the process and status information
+* The connection to the migration service hosted at **console.redhat.com**
+
+If you can reach the login page but the source is still not appearing, check the agent’s connection status:
+
+Navigate to: **`https://<machine-ip>:3333/api/v1/status`**.  
+You will likely see output similar to:
+
+```json
+{
+"status": "waiting-for-credentials",
+"connected": "false",
+"statusInfo": "No credentials provided"
+}
+```
+
+If the output indicates that the service is unreachable, follow these steps:  
+1. Review the network traffic from the agent to console.redhat.com.
+2. Ensure that the necessary outbound connections are allowed in the firewall.
```

Comment:
I would also somewhere tell the user he need to re-check his network assigned to network inerface of the VM, if it is proper network that has connection to public internet.